src/test/java/it/aboutbits/springboot/toolbox/persistence/javatype/WrapperTypesJpaTest.java

```diff
@@ -30,308 +39,620 @@ public class WrapperTypesJpaTest {
+                        .usingRecursiveComparison()
+                        .isEqualTo(savedItem);
+            }
 
-        @ParameterizedTest
-        @ValueSource(doubles = {-1, 0, 1, -0.001, 0.001, -100_000_000, 100_000_000})
-        void givenValues_inAndOut_shouldSucceed(double doubleValue) {
-            var item = new WrapperTypesModel();
-            item.setBigDecimalValue(new WrapBigDecimal(BigDecimal.valueOf(doubleValue)));
+            @ParameterizedTest
+            @ValueSource(ints = {-1, 0, 1, -100_000_000, 100_000_000})
+            void givenValues_inAndOut_shouldSucceed(int intValue) {
+                var item = new WrapperTypesModel();
+                item.setBigIntegerValue(new WrapBigIntegerRecord(BigInteger.valueOf(intValue)));
 
-            var savedItem = repository.save(item);
+                var savedItem = repository.save(item);
 
-            var retrievedItem = repository.findByBigDecimalValue(savedItem.getBigDecimalValue());
+                var retrievedItem = repository.findByBigIntegerValue(savedItem.getBigIntegerValue());
 
-            assertThat(retrievedItem).isPresent()
-                    .get()
-                    .usingRecursiveComparison()
-                    .isEqualTo(savedItem);
+                assertThat(retrievedItem).isPresent()
+                        .get()
+                        .usingRecursiveComparison()
+                        .isEqualTo(savedItem);
+            }
         }
-    }
 
-    @Nested
-    class WrapBigIntegerType {
-        @Test
-        void givenNull_inAndOut_shouldSucceed() {
-            var item = new WrapperTypesModel();
-            item.setBigIntegerValue(null);
+        @Nested
+        class WrapDoubleRecordType {
+            @Test
+            void givenNull_inAndOut_shouldSucceed() {
+                var item = new WrapperTypesModel();
+                item.setDoubleValue(null);
 
-            var savedItem = repository.save(item);
+                var savedItem = repository.save(item);
 
-            var retrievedItem = repository.findByBigIntegerValue(null);
+                var retrievedItem = repository.findByDoubleValue(null);
 
-            assertThat(retrievedItem).isPresent()
-                    .get()
-                    .usingRecursiveComparison()
-                    .isEqualTo(savedItem);
-        }
+                assertThat(retrievedItem).isPresent()
+                        .get()
+                        .usingRecursiveComparison()
+                        .isEqualTo(savedItem);
+            }
 
-        @ParameterizedTest
-        @ValueSource(ints = {-1, 0, 1, -100_000_000, 100_000_000})
-        void givenValues_inAndOut_shouldSucceed(int intValue) {
-            var item = new WrapperTypesModel();
-            item.setBigIntegerValue(new WrapBigInteger(BigInteger.valueOf(intValue)));
+            @ParameterizedTest
+            @ValueSource(doubles = {-1, 0, 1, -0.001, 0.001, -100_000_000, 100_000_000})
```

Comment:
You could add the `d` suffix here same as you did it for float, but it won't make a real difference:
```suggestion
            @ValueSource(doubles = {-1, 0, 1, -0.001d, 0.001d, -100_000_000, 100_000_000})
```